test/cmdlineTests/optimizer_array_sload/output

```diff
@@ -0,0 +1,83 @@
+Optimized IR:
+/*******************************************************
+ *                       WARNING                       *
+ *  Solidity to Yul compilation is still EXPERIMENTAL  *
+ *       It can result in LOSS OF FUNDS or worse       *
+ *                !USE AT YOUR OWN RISK!               *
+ *******************************************************/
+
+object "Arraysum_32" {
+    code {
+        {
+            mstore(64, 128)
+            if callvalue() { revert(0, 0) }
+            let _1 := datasize("Arraysum_32_deployed")
+            codecopy(0, dataoffset("Arraysum_32_deployed"), _1)
+            return(0, _1)
+        }
+    }
+    object "Arraysum_32_deployed" {
+        code {
+            {
+                mstore(64, 128)
+                if iszero(lt(calldatasize(), 4))
+                {
+                    let _1 := 0
+                    if eq(0x81d73423, shr(224, calldataload(_1)))
+                    {
+                        if callvalue() { revert(_1, _1) }
+                        if slt(add(calldatasize(), not(3)), _1) { revert(_1, _1) }
+                        let vloc_sum := _1
+                        let vloc_i := _1
+                        let _2 := sload(_1)
+                        for { }
+                        lt(vloc_i, _2)
+                        {
+                            vloc_i := increment_t_uint256(vloc_i)
+                        }
+                        {
+                            let _3, _4 := storage_array_index_access_t_array$_t_uint256_$dyn_storage(_1, vloc_i)
+                            vloc_sum := checked_add_t_uint256(vloc_sum, extract_from_storage_value_dynamict_uint256(sload(_3), _4))
+                        }
```

Comment:
[Diff](https://gist.github.com/hrkrshnn/5d054ed71e068fbd02e9ead9599a6af9) to illustrate the change.

Reply:
W ereally need to fix the inliner...